storage/badger/chunk_consumer.go

```diff
@@ -0,0 +1,51 @@
+package badger
+
+import (
+	"fmt"
+
+	"github.com/dgraph-io/badger/v2"
+
+	"github.com/onflow/flow-go/storage/badger/operation"
+)
+
+const (
+	JobConsumerChunk = "ChunkConsumer"
+)
+
+type ChunkConsumer struct {
+	db *badger.DB
+}
+
+func NewChunkConsumer(db *badger.DB) *ChunkConsumer {
+	return &ChunkConsumer{
+		db: db,
+	}
+}
+
+func (cc *ChunkConsumer) ProcessedIndex() (int64, error) {
+	var processed int64
+	err := cc.db.View(operation.RetrieveProcessedIndex(JobConsumerChunk, &processed))
+	if err != nil {
+		return 0, fmt.Errorf("could not retrieve processed index: %w", err)
+	}
+	return processed, nil
+}
+
+func (cc *ChunkConsumer) InitProcessedIndex() (int64, error) {
+	initprocessed := int64(0)
```

Comment:
the default processing index is `0`, it also assumes job queue will store jobs starting from index `0`